### PR TITLE
3125 Moved fips-county table to the database

### PR DESCRIPTION
**Pull request by carlosfelix2**
This PR moves the FIPS county table to the database as a first live test of the lookup tables.  Once we are happy with this change we will migrate the rest of the tables.  

Moving the tables to the database has resulted in the Metadata class requiring the tables to be already in the database BEFORE using the metadata, and the metadata instance is being used pretty much everywhere.  That meant that unit and integration tests fail due to the tables not being present in the database with a new or reset database (e.g. in Github Actions).  To solve this issue, I changed the unit tests to use a simple metadata instance for those tests that could, but some tests required the full metadata catalog which in turn does not make them unit tests and they were moved to the integration tests suite.  I also changed some code to allow for metadata injection for unit tests.  NOTE THAT THE UNIT AND INTEGRATION TESTS HAVE NOT BEEN CREATED OR CHANGED BESIDES CHANGING THE METADATA, so focus your review on the other changes.

**Test Steps:**
1. Build this branch with `./gradlew package`
2. Run your local docker container with `docker compose up`.  For those that cannot run the docker container, you must run `./gradlew run` then run `./gradlew reloadTables` and `./gradlew reloadSettings` once the functions are running.
3. Run the smoke tests and verify all tests pass.
4. Run `./gradlew testIntegration` and verify the tests pass.
5. IMPORTANT: Verify that the fips-county table has been uploaded to staging and production by running `./prime lookuptables list` for EACH of those environments.  The idea here is to make sure the tables are loaded before deploying this branch.

## Changes
- moved the FIPS county table to the database
- Allowed for injection of metadata
- Changed the use of metadata in the unit tests
- Moved the integration tests to run after the container is running
- Moved some unit tests to the integration test suite

## Checklist

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [x] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Fixes
- #3125

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

## Pull reviewers stats
Stats for the last 30 days:
|                                                                                                                                                                            | User               | Total reviews | Median time to review                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | Total comments |
| -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------ | ------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------- |
| <a href="https://github.com/MauriceReeves-usds"><img src="https://avatars.githubusercontent.com/u/74194189?u=f81b3f069b28469e1fb39322ba3c17112c1dd1d0&v=4" width="20"></a> | MauriceReeves-usds | **33**        | [19h 27m](https://app.flowwer.dev/charts/review-time/~(u~(i~'74194189~n~'MauriceReeves-usds)~p~30~r~(~(d~'r23qbd~t~'2t)~(d~'r21xa7~t~'4e9)~(d~'r29e6y~t~'5k3y)~(d~'r29e4s~t~'4viy)~(d~'r29e5m~t~'4zrn)~(d~'r29e8i~t~'78zv)~(d~'r2pzyo~t~'15o0)~(d~'r2o37b~t~'27q)~(d~'r2doob~t~'dpl)~(d~'r2rual~t~'1pbs)~(d~'r2mluh~t~'b6ge)~(d~'r2mlha~t~'5ui)~(d~'r29e9q~t~'5d81)~(d~'r2u76d~t~'1ab)~(d~'r2gqvu~t~'62hn)~(d~'r316mf~t~'1sao)~(d~'r2zf3l~t~'52cm)~(d~'r2z8ca~t~'4zmi)~(d~'r2zb63~t~'aqke)~(d~'r2gqti~t~'1i17)~(d~'r2gqsv~t~'17n5)~(d~'r2mlni~t~'58nc)~(d~'r2gqu9~t~'3cdy)~(d~'r3elo4~t~'bd)~(d~'r3el16~t~'7w)~(d~'r3em56~t~'le)~(d~'r3d4ef~t~'jc2)~(d~'r3d52y~t~'a8c)~(d~'r3eavf~t~'15m8)~(d~'r3d5z9~t~'6e7a)~(d~'r3ekss~t~'qv)~(d~'r3emzs~t~'24g)~(d~'r3d55g~t~'5zzo))))                                                                                                                                                                                                                                                       | 6              |
| <a href="https://github.com/cwinters-usds"><img src="https://avatars.githubusercontent.com/u/86251310?u=e7f46bd48ca250dea43f9e0d6433a7fb8627ed30&v=4" width="20"></a>      | cwinters-usds      | 26            | [14h 12m](https://app.flowwer.dev/charts/review-time/~(u~(i~'86251310~n~'cwinters-usds)~p~30~r~(~(d~'r29gtt~t~'3edm)~(d~'r2ruul~t~'101)~(d~'r2qefq~t~'f6)~(d~'r2qeq5~t~'pl)~(d~'r2qerc~t~'qs)~(d~'r2muxy~t~'kchs)~(d~'r2mv74~t~'kcqy)~(d~'r2mv8s~t~'kcsm)~(d~'r2q773~t~'cw8d)~(d~'r2rzdk~t~'13e4)~(d~'r2schs~t~'1pa)~(d~'r2pzle~t~'5at)~(d~'r2pzr3~t~'5gi)~(d~'r2tp4t~t~'3k60)~(d~'r2mulc~t~'314)~(d~'r2mum0~t~'31s)~(d~'r2muoq~t~'34i)~(d~'r2mty3~t~'ebb)~(d~'r2mwwq~t~'bcmi)~(d~'r2mwxj~t~'9nyb)~(d~'r2mwzl~t~'9kua)~(d~'r2of07~t~'9o)~(d~'r2zet7~t~'3jw)~(d~'r2mu1x~t~'1ip)~(d~'r2h370~t~'5j)~(d~'r2h24e~t~'19n)~(d~'r2d69n~t~'18d)~(d~'r2d6md~t~'1gh)~(d~'r2zet7~t~'3jw)~(d~'r2qedb~t~'3ua)~(d~'r3chws~t~'4wy)~(d~'r33f07~t~'ffd)~(d~'r2zf0k~t~'13gs)~(d~'r2zf23~t~'13ib)~(d~'r2zfk1~t~'1409)~(d~'r2zfm7~t~'142f)~(d~'r2zfvq~t~'14by)~(d~'r2zgar~t~'14qz)~(d~'r2zgja~t~'14zi)~(d~'r2zgko~t~'150w)~(d~'r2zgor~t~'154z)~(d~'r3hwfo~t~'1a3q)~(d~'r3hwgo~t~'1a4q)~(d~'r3hwhi~t~'1a5k)~(d~'r3hww8~t~'1aka)~(d~'r3hwzh~t~'1anj)))) | 19             |
| <a href="https://github.com/kevinhaube"><img src="https://avatars.githubusercontent.com/u/4022304?u=c0f80e2ad03386621394d1830dc33b7b5cc23680&v=4" width="20"></a>          | kevinhaube         | 21            | [1h 5m](https://app.flowwer.dev/charts/review-time/~(u~(i~'4022304~n~'kevinhaube)~p~30~r~(~(d~'r21wn6~t~'nqo)~(d~'r21ws4~t~'nvm)~(d~'r224pg~t~'1r3)~(d~'r229x5~t~'311)~(d~'r2sf3m~t~'a)~(d~'r2sen7~t~'6p)~(d~'r2seuw~t~'e)~(d~'r24598~t~'1ot)~(d~'r2zezt~t~'u)~(d~'r31f6z~t~'4c4)~(d~'r2zjny~t~'3q)~(d~'r2ziqy~t~'u)~(d~'r2zjjk~t~'c5)~(d~'r2gqgl~t~'3diq)~(d~'r2gq9s~t~'354l)~(d~'r2h5mm~t~'1o)~(d~'r2h5ea~t~'5q)~(d~'r2h33t~t~'2c)~(d~'r2h267~t~'1bg)~(d~'r2mrpn~t~'2p)~(d~'r31f6z~t~'4c4)~(d~'r3hsru~t~'15g7)~(d~'r3i17u~t~'1dw7)~(d~'r3i18e~t~'1dwr)~(d~'r3jmvl~t~'17b5)~(d~'r3htfd~t~'173f)~(d~'r3i0jd~t~'34w)~(d~'r3jn0b~t~'12bb)~(d~'r3jn5z~t~'12gz))))                                                                                                                                                                                                                                                                                                                                                                   | 12             |
| <a href="https://github.com/carlosfelix2"><img src="https://avatars.githubusercontent.com/u/63804190?u=009d26bf9914817d2f9a066811541d0f6f3155c9&v=4" width="20"></a>       | carlosfelix2       | 19            | [10h 23m](https://app.flowwer.dev/charts/review-time/~(u~(i~'63804190~n~'carlosfelix2)~p~30~r~(~(d~'r23xe1~t~'2hjv)~(d~'r23x1g~t~'29z)~(d~'r2di02~t~'1tl)~(d~'r2dim1~t~'2fk)~(d~'r2mc1j~t~'1z0g)~(d~'r2q06x~t~'g)~(d~'r2qcc0~t~'1ihh)~(d~'r2rv2m~t~'182)~(d~'r2rv2q~t~'186)~(d~'r2ml48~t~'9a5i)~(d~'r2q97e~t~'cy8o)~(d~'r2o5zo~t~'2e)~(d~'r2d2wj~t~'2eic)~(d~'r2d2yd~t~'2ek6)~(d~'r23wq2~t~'1l79)~(d~'r246pr~t~'s6)~(d~'r246z2~t~'11h)~(d~'r2b5zo~t~'17n7)~(d~'r2d4z4~t~'fl)~(d~'r2d66q~t~'10u)~(d~'r2scre~t~'228d)~(d~'r2sdwg~t~'23df)~(d~'r3e8nf~t~'19a)~(d~'r337mz~t~'asd)~(d~'r337on~t~'au1)~(d~'r3387t~t~'bd7)~(d~'r33f6c~t~'22v)~(d~'r33f74~t~'23n)~(d~'r33fa8~t~'26r)~(d~'r33fbo~t~'287)~(d~'r314i9~t~'e1s5)~(d~'r3hsnf~t~'3elw)~(d~'r2tw63~t~'1eog)~(d~'r2tw6x~t~'1epa)~(d~'r2tw9t~t~'1es6)~(d~'r2tweg~t~'1ewt)~(d~'r2twfc~t~'1exp)~(d~'r2twj3~t~'1f1g)~(d~'r3g78r~t~'e1c)~(d~'r3cahz~t~'55c7))))                                                                                                                        | **65**         |
| <a href="https://github.com/clediggins-usds"><img src="https://avatars.githubusercontent.com/u/89804087?u=5b03415b2bc7766f96908dd40f522633187e4b02&v=4" width="20"></a>    | clediggins-usds    | 17            | [2h 16m](https://app.flowwer.dev/charts/review-time/~(u~(i~'89804087~n~'clediggins-usds)~p~30~r~(~(d~'r2osyj~t~'28zv)~(d~'r2otta~t~'2dcn)~(d~'r2qm4i~t~'91u)~(d~'r2qmoh~t~'fk)~(d~'r2movr~t~'b9ho)~(d~'r2moys~t~'b9kp)~(d~'r2mrtc~t~'4h)~(d~'r2oq3n~t~'ch)~(d~'r2opoi~t~'ks)~(d~'r2op0l~t~'fd)~(d~'r2oo1i~t~'6ry)~(d~'r2oojb~t~'fk)~(d~'r2u161~t~'1mo)~(d~'r2zqnb~t~'5kr9)~(d~'r29jy1~t~'7i0y)~(d~'r29k35~t~'1l)~(d~'r29k4r~t~'37)~(d~'r2bni2~t~'5uv)~(d~'r2dmpd~t~'93)~(d~'r2dncq~t~'iz)~(d~'r2dme6~t~'48du)~(d~'r3csl7~t~'9dph)~(d~'r3ckty~t~'8u3y)~(d~'r2twb8~t~'1etl))))                                                                                                                                                                                                                                                                                                                                                                                                                                                     | 31             |
| <a href="https://github.com/sean-usds"><img src="https://avatars.githubusercontent.com/u/87096393?u=90127e88c4ff845b6a754d10151b10a5158562ff&v=4" width="20"></a>          | sean-usds          | 16            | [1h 56m](https://app.flowwer.dev/charts/review-time/~(u~(i~'87096393~n~'sean-usds)~p~30~r~(~(d~'r20lh2~t~'1a6)~(d~'r2twyx~t~'3k0q)~(d~'r2u7k7~t~'w6)~(d~'r2ugvm~t~'37v)~(d~'r2dcq1~t~'14z5)~(d~'r2mhsn~t~'971y)~(d~'r2gwsm~t~'48n)~(d~'r2btm6~t~'46l)~(d~'r2d6of~t~'1ij)~(d~'r2dc5j~t~'7m)~(d~'r3ecas~t~'21o5)~(d~'r3gcz2~t~'42fy)~(d~'r3ebcm~t~'cil5)~(d~'r3ecl0~t~'2200)~(d~'r3ecg5~t~'21ua)~(d~'r3gsti~t~'6hk)~(d~'r3cppe~t~'lj)~(d~'r3cpsw~t~'p1))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | 4              |
| <a href="https://github.com/jimduff-usds"><img src="https://avatars.githubusercontent.com/u/62258514?u=b7ce81d1478c472fb5bf6f3d31edb1a4454d55dd&v=4" width="20"></a>       | jimduff-usds       | 15            | [17h 23m](https://app.flowwer.dev/charts/review-time/~(u~(i~'62258514~n~'jimduff-usds)~p~30~r~(~(d~'r2puoy~t~'10ea)~(d~'r23zoi~t~'4x1)~(d~'r29u0m~t~'5eac)~(d~'r2tkqx~t~'1kah)~(d~'r2tysp~t~'3ttw)~(d~'r2movm~t~'b9hj)~(d~'r2mp63~t~'9nme)~(d~'r2touj~t~'16j)~(d~'r2tkxk~t~'302l)~(d~'r2n47j~t~'ho)~(d~'r2tkxk~t~'302l)~(d~'r322cl~t~'dp8)~(d~'r330n0~t~'3se)~(d~'r311rv~t~'12ef)~(d~'r313ma~t~'148u)~(d~'r3ggw4~t~'4t4)~(d~'r3i91h~t~'211c)~(d~'r3i7xv~t~'5lj3))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | 17             |
| <a href="https://github.com/RickHawesUSDS"><img src="https://avatars.githubusercontent.com/u/48692522?u=323d76e6f867d8ed0b3e56d7e5808313bfcc2ccf&v=4" width="20"></a>      | RickHawesUSDS      | 13            | [4h 17m](https://app.flowwer.dev/charts/review-time/~(u~(i~'48692522~n~'RickHawesUSDS)~p~30~r~(~(d~'r2ou7r~t~'8m)~(d~'r2rvbi~t~'1w6)~(d~'r2mt2k~t~'haa)~(d~'r2s8yz~t~'f8i)~(d~'r2uatn~t~'ei)~(d~'r31dof~t~'4ve)~(d~'r2ow0j~t~'beu)~(d~'r2owa9~t~'bok)~(d~'r2zepb~t~'12x)~(d~'r2zfft~t~'5f79)~(d~'r3ceor~t~'cxsc)~(d~'r3cspw~t~'j2n)~(d~'r31qgm~t~'9bz)~(d~'r318o4~t~'2lz)~(d~'r31d8f~t~'vr)~(d~'r31tz6~t~'bw1)~(d~'r3cwio~t~'9hmy)~(d~'r3cwwy~t~'9i18)~(d~'r2qq41~t~'idj)~(d~'r3cbt3~t~'5fn2)~(d~'r3i1c3~t~'3o8m))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | 17             |
| <a href="https://github.com/TomNUSDS"><img src="https://avatars.githubusercontent.com/u/74203452?u=6eccca154e9d5c808c5269d9604648aec3c9a412&v=4" width="20"></a>           | TomNUSDS           | 9             | [1h 28m](https://app.flowwer.dev/charts/review-time/~(u~(i~'74203452~n~'TomNUSDS)~p~30~r~(~(d~'r22mrf~t~'4b9)~(d~'r22msa~t~'4c4)~(d~'r2zq07~t~'o)~(d~'r2zrhe~t~'3m)~(d~'r22mhg~t~'fgj)~(d~'r241bx~t~'1ub0)~(d~'r2dek4~t~'3tf)~(d~'r2gya9~t~'11)~(d~'r2mldj~t~'6tq)~(d~'r2mlfq~t~'6vx)~(d~'r2mr1r~t~'1tw)~(d~'r2bqfo~t~'103)~(d~'r31hlm~t~'gz)~(d~'r3hyei~t~'1b2v))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | 7              |
| <a href="https://github.com/whytheplatypus"><img src="https://avatars.githubusercontent.com/u/410846?v=4" width="20"></a>                                                  | whytheplatypus     | 7             | [2h 23m](https://app.flowwer.dev/charts/review-time/~(u~(i~'410846~n~'whytheplatypus)~p~30~r~(~(d~'r2qel8~t~'ko)~(d~'r2rwr2~t~'fx)~(d~'r2q0xk~t~'6mz)~(d~'r2tm3y~t~'1pu)~(d~'r224er~t~'2ho)~(d~'r29g58~t~'7e85)~(d~'r3c5du~t~'91uk)~(d~'r3e0ib~t~'3hfp)~(d~'r2tsy0~t~'1bgd)~(d~'r3g87k~t~'7f)~(d~'r3i8g4~t~'20fz))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | 10             |
| <a href="https://github.com/jeremy-page"><img src="https://avatars.githubusercontent.com/u/54326889?u=600f8a60859e9b0ca4634227f00b2cd9bee5e9b8&v=4" width="20"></a>        | jeremy-page        | 4             | [14m](https://app.flowwer.dev/charts/review-time/~(u~(i~'54326889~n~'jeremy-page)~p~30~r~(~(d~'r23x3c~t~'16d)~(d~'r23pax~t~'r7)~(d~'r246w1~t~'j5)~(d~'r244b7~t~'74))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | 0              |
| <a href="https://github.com/brick-green-agile6"><img src="https://avatars.githubusercontent.com/u/86254221?u=894ef5a4773dd01e92be595af8f1879041f85002&v=4" width="20"></a> | brick-green-agile6 | 4             | [13h 50m](https://app.flowwer.dev/charts/review-time/~(u~(i~'86254221~n~'brick-green-agile6)~p~30~r~(~(d~'r2bqtd~t~'12f6)~(d~'r2zurg~t~'9hez)~(d~'r2moeo~t~'jr)~(d~'r2zurg~t~'9hez)~(d~'r33c6l~t~'pb))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | 1              |
| <a href="https://github.com/ahay-agile6"><img src="https://avatars.githubusercontent.com/u/19178435?u=aa7d4ddd31665f83e677e20967502557741345e4&v=4" width="20"></a>        | ahay-agile6        | 2             | [1d 9h 28m](https://app.flowwer.dev/charts/review-time/~(u~(i~'19178435~n~'ahay-agile6)~p~30~r~(~(d~'r2mtil~t~'62o)~(d~'r2zdip~t~'4zuy))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | 0              |
| <a href="https://github.com/ronaldheft-usds"><img src="https://avatars.githubusercontent.com/u/52176593?u=be6fa1049b61315b4ab5ba3189e9f111f726db5d&v=4" width="20"></a>    | ronaldheft-usds    | 2             | [**9m**](https://app.flowwer.dev/charts/review-time/~(u~(i~'52176593~n~'ronaldheft-usds)~p~30~r~(~(d~'r22bw6~t~'fn)~(d~'r22bxa~t~'gr)~(d~'r220wp~t~'4f))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | 1              |
| <a href="https://github.com/jbiskie"><img src="https://avatars.githubusercontent.com/u/84460447?v=4" width="20"></a>                                                       | jbiskie            | 2             | [1h 5m](https://app.flowwer.dev/charts/review-time/~(u~(i~'84460447~n~'jbiskie)~p~30~r~(~(d~'r23rf5~t~'16l)~(d~'r3g187~t~'4sm))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | 0              |
| <a href="https://github.com/sethdarragile6"><img src="https://avatars.githubusercontent.com/u/92405130?u=b36435069928f7e5f03109c9f478cdc7158fc01e&v=4" width="20"></a>     | sethdarragile6     | 2             | [17h 34m](https://app.flowwer.dev/charts/review-time/~(u~(i~'92405130~n~'sethdarragile6)~p~30~r~(~(d~'r245z0~t~'1ug7)~(d~'r245za~t~'1ugh)~(d~'r245zg~t~'1ugn)~(d~'r249ow~t~'1y63)~(d~'r24exs~t~'907)~(d~'r24eyd~t~'90s)~(d~'r24eyn~t~'912)~(d~'r24f5c~t~'97r)~(d~'r29jrp~t~'54vj)~(d~'r2bb5a~t~'1cst)~(d~'r2gywn~t~'6co))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | 9              |
| <a href="https://github.com/JosiahSiegel"><img src="https://avatars.githubusercontent.com/u/5522990?v=4" width="20"></a>                                                   | JosiahSiegel       | 1             | [5d 2h 55m](https://app.flowwer.dev/charts/review-time/~(u~(i~'5522990~n~'JosiahSiegel)~p~30~r~(~(d~'r2ms4x~t~'9h67)~(d~'r2mso5~t~'9hpf))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | 2              |
| <a href="https://github.com/MikeC-A6"><img src="https://avatars.githubusercontent.com/u/77452595?v=4" width="20"></a>                                                      | MikeC-A6           | 1             | [7h 19m](https://app.flowwer.dev/charts/review-time/~(u~(i~'77452595~n~'MikeC-A6)~p~30~r~(~(d~'r3g954~t~'19c4)~(d~'r3hpr9~t~'j8f)~(d~'r3hqu5~t~'kbb))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | 3              |
| <a href="https://github.com/rhood23699"><img src="https://avatars.githubusercontent.com/u/86322826?v=4" width="20"></a>                                                    | rhood23699         | 1             | [14h 10m](https://app.flowwer.dev/charts/review-time/~(u~(i~'86322826~n~'rhood23699)~p~30~r~(~(d~'r3ft6z~t~'13ca))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | 0              |

### CODE REVIEW

File: .github/workflows/build_hub.yml [hunk 1]
```diff
@@ -106,7 +91,26 @@ jobs:
           docker-compose -f docker-compose.yml up -d prime_dev redox ftps sftp azurite
 
       - name: Load settings
-        run: docker-compose -f docker-compose.yml up settings
+        run: ./gradlew reloadSettings
```
Comment (carlosfelix2) [inline .github/workflows/build_hub.yml:94]: Moved the integration tests to run after the container is running and added reloadTables too.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -47,21 +47,6 @@ jobs:
           # This path is from the root of the repo as needed by the plugin
           files: prime-router/build/test-results/test/**/*.xml
 
-      - name: Run Integration Tests
```
Comment (carlosfelix2) [inline .github/workflows/release.yml:50]: Context: release build is not running containers or smoke tests, so integration tests need to be removed.

File: prime-router/src/main/kotlin/azure/WorkflowEngine.kt [hunk 3]
```diff
@@ -45,27 +45,55 @@ import java.time.OffsetDateTime
...
+    val sftpTransport: SftpTransport = SftpTransport()
+    val redoxTransport: RedoxTransport = RedoxTransport()
+    val as2Transport: AS2Transport = AS2Transport()
+    val ftpsTransport: FTPSTransport = FTPSTransport()
+
+    /**
+     * Instantiate a workflow engine instance with optional [metadata], [settings], [db], [blob] and [queue]
+     * instances used for dependency injection.
+     */
+    constructor(
```
Comment (carlosfelix2) [inline prime-router/src/main/kotlin/azure/WorkflowEngine.kt:67]: Context: Changed the constructor, so we can inject metadata for tests.

File: prime-router/src/main/kotlin/azure/WorkflowEngine.kt [hunk 4]
```diff
@@ -702,21 +730,14 @@ class WorkflowEngine(
          * These are all potentially heavy weight objects that
          * should only be created once.
          */
-        private val metadata = Metadata.getInstance()
+        private val metadata by lazy { Metadata.getInstance() }
```
Comment (carlosfelix2) [inline prime-router/src/main/kotlin/azure/WorkflowEngine.kt:733]: Context: Adding lazy here allows for using an injected metadata and not having a real metadata catalog being initialized if not needed.

File: prime-router/src/test/kotlin/metadata/LookupTableTests.kt [hunk 5]
```diff
@@ -105,28 +104,6 @@ class LookupTableTests {
         assertThat(value.isNullOrEmpty()).isTrue()
     }
 
-    @Test
-    fun `test zip code lookup`() {
```
Comment (carlosfelix2) [inline prime-router/src/test/kotlin/metadata/LookupTableTests.kt:109]: Context:  Not a unit tests, plus this is tested in the integration tests

File: prime-router/src/test/kotlin/unittest/UnitTestUtils.kt [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        val simpleSchema = Schema(name = "one", topic = "test", elements = listOf(Element("a"), Element("b")))
+
+        /**
+         * A simple metadata instance that does not use the real configuration.
+         */
+        val simpleMetadata by lazy {
+            Metadata(simpleSchema)
+        }
+
+        @Deprecated(
```
Comment (carlosfelix2) [inline prime-router/src/test/kotlin/unittest/UnitTestUtils.kt:24]: Note the deprecated here.  We need to move or redo any tests that require a full metadata catalog.  Once we can do that we need to remove this variable and the other metadata files this uses.

File: .github/workflows/build_hub.yml [hunk 7]
```diff
@@ -103,10 +88,26 @@ jobs:
 
       - name: Spin up containers
         run: |
-          docker-compose -f docker-compose.yml up -d prime_dev redox ftps sftp azurite
+          docker-compose -f docker-compose.yml up -d prime_dev redox ftps sftp azurite settings
```
Comment (carlosfelix2) [inline .github/workflows/build_hub.yml:91]: Context: gradle reloadSettings does not have a wait function yet, so we must use the settings container to make sure we do not get a failure.  See #3271

File: .github/workflows/build_hub.yml [hunk 8]
```diff
@@ -124,6 +125,55 @@ jobs:
           rm -f .gradle/caches/modules-2/modules-2.lock
           rm -f .gradle/caches/modules-2/gc.properties
 
+      - name: Generate New Docs
+        run: |
+          # Clean the docs before regenerating
```
Comment (carlosfelix2) [inline .github/workflows/build_hub.yml:130]: Context: This also needs the full metadata catalog, but this check will go away as soon as we migrate the schemas to the database.

File: prime-router/src/test/kotlin/FakeReportTests.kt [hunk 9]
```diff
@@ -195,42 +195,4 @@ class FakeReportTests {
         val expected = "Any lab USA, Any facility USA"
         assertThat(actual).isEqualTo(expected)
     }
-
-    @Test
```
Comment (carlosfelix2) [inline prime-router/src/test/kotlin/FakeReportTests.kt:199]: Context: Tests that require the full metadata catalog were moved to be integration tests

File: .environment/gitleaks/gitleaks-config.toml [hunk 10]
```diff
@@ -156,7 +156,7 @@ title = "PRIME ReportStream Gitleaks Configuration"
             'API_ENDPOINT_HOST\"\) \?: \"localhost\"',
             'api_url:',
             'apiKey: \"some_key\"',
-            'apiKey: \"some_key\"',
+            'API_KEY = "x-api-key"',
```
Comment (carlosfelix2) [inline .environment/gitleaks/gitleaks-config.toml:159]: Context: Not my code changes, but gitleaks was complaining.

File: .github/workflows/build_hub.yml [hunk 11]
```diff
@@ -61,7 +61,7 @@ jobs:
         run: docker-compose -f docker-compose.postgres.yml up -d
 
       - name: Build Prime Router Package
-        run: ./gradlew package -x fatjar -Pshowtests
+        run: ./gradlew package -Pshowtests
```
Comment (carlosfelix2) [inline .github/workflows/build_hub.yml:64]: Context: fatjar is needed now due to loading the tables in start_func.sh

File: .github/workflows/build_hub.yml [hunk 12]
```diff
@@ -74,21 +74,6 @@ jobs:
           # This path is from the root of the repo as needed by the plugin
           files: prime-router/build/test-results/test/**/*.xml
 
-      - name: Run Integration Tests
```
Comment (carlosfelix2) [inline .github/workflows/build_hub.yml:77]: Context: integration tests require the full metadata catalog, which needs all settings and tables, which needs the API running, which needs the containers running

File: .github/workflows/build_hub.yml [hunk 13]
```diff
@@ -232,3 +186,50 @@ jobs:
         if: ${{ steps.check_changes.outcome == 'failure' }}
         run: |
           false
+
+  docker_build_test:
```
Comment (carlosfelix2) [inline .github/workflows/build_hub.yml:190]: Context: Not new, just moved

File: .github/workflows/build_hub.yml [hunk 14]
```diff
@@ -232,3 +186,50 @@ jobs:
...
+          
+  linting:
+    name: Check Linting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Deprecated step always succeeds
+        run: true
+        
+  docs:
+    # Deprecated job.  Check schema docs was moved to build_router job
```
Comment (carlosfelix2) [inline .github/workflows/build_hub.yml:229]: Context: Will need to remove this once the check in Github is removed and everyone merge all these changes from master.

File: .github/workflows/build_hub.yml [hunk 15]
```diff
@@ -116,6 +126,12 @@ jobs:
           export $(xargs < .vault/env/.env.local)
           ./prime create-credential --type=UserPass --persist=DEFAULT-SFTP --user foo --pass pass
           ./gradlew testSmoke
+
```
Comment (cwinters-usds) [inline .github/workflows/build_hub.yml:129]: is this for debugging or meant to be permanent? (just curious, I don't think either is incorrect/wrong)

**Review by cwinters-usds — APPROVED** [review]
Changes to gitleaks and GH workflow look good to me. :tada:

**Review by jimduff-usds — APPROVED** [review]
I grabbed the code, deleted my database, and when I start docker it loads the tables:
```
prime_dev_1        | [LOAD TABLES] Loading 1 tables from /prime-data-hub-router/azure-functions/prime-data-hub-router/../../resources/test/metadata/tables...
prime_dev_1        | [LOAD TABLES] Creating table fips-county...
prime_dev_1        | [LOAD TABLES] 3232 rows created for lookup table fips-county version 1.
prime_dev_1        | [LOAD TABLES] Table version 1 is now active.
prime_dev_1        | [LOAD TABLES] Done.

```


No more chicken n egg problem!

Also reviewed during the meeting.

File: prime-router/src/main/kotlin/Report.kt [hunk 16]
```diff
@@ -277,7 +277,8 @@ class Report : Logging {
         sources: List<Source>,
         destination: Receiver? = null,
         bodyFormat: Format? = null,
-        itemLineage: List<ItemLineage>? = null
+        itemLineage: List<ItemLineage>? = null,
+        metadata: Metadata? = null
```
Comment (carlosfelix2) [inline prime-router/src/main/kotlin/Report.kt:281]: Try to default the metadata at the highest level possible.

File: prime-router/src/main/kotlin/Report.kt [hunk 17]
```diff
@@ -287,7 +288,7 @@ class Report : Logging {
         this.bodyFormat = bodyFormat ?: destination?.format ?: Format.INTERNAL
         this.itemLineages = itemLineage
         this.createdDateTime = OffsetDateTime.now()
-        this.metadata = Metadata.getInstance()
+        this.metadata = metadata ?: Metadata.getInstance()
```
Comment (whytheplatypus) [inline prime-router/src/main/kotlin/Report.kt:291]: For a follow up ticket: The `Report` class doesn't actually need metadata. It looks like it's only used by some fake data generation which should be a separate concern. Extracting that behavior would simplify things.

File: .github/workflows/build_hub.yml (hunk 15 above)
Comment (carlosfelix2) [reply .github/workflows/build_hub.yml:129]: Permanent.  This will make it easier to see why something fails.

**Review by MauriceReeves-usds — APPROVED** [review]

**Pull request merged**